### PR TITLE
Improvements around ad username vs email and basic auth vs token auth

### PR DESCRIPTION
**Pull request by elliottzack429**
## Change Log Items

* Alks developer configure no longer collects userID unless basic auth selected

## Description

Adds some hints around the network username field.  Also changes developer configure to only get the userid id if basic auth is selected

### CODE REVIEW

File: src/lib/promptForUserId.ts [hunk 1]
```diff
@@ -5,5 +5,14 @@ export async function promptForUserId(): Promise<string> {
   // Ignore failure since we're about to prompt for it
   const userId = await getUserId();
 
-  return getPrompt('userid', userId, 'Network Username', null);
+  return getPrompt(
+    'userid',
+    userId,
+    'Active Directory Username (this is not part of your email)',
+    null
+  );
```
Comment (LumaC0) [inline src/lib/promptForUserId.ts:13]: It's not always clear to users where they can find this. Maybe in the documentation we can say something about finding this information on the ALKS users tab.

File: src/lib/handlers/alks-developer-configure.ts [hunk 2]
```diff
@@ -63,6 +59,10 @@ export async function handleAlksDeveloperConfigure(
           break;
         }
         case PASSWORD_AUTH_CHOICE: {
+          if (options.username || shouldPrompt) {
+            await setUserId(options.username ?? (await promptForUserId()));
+          }
+
```
[the remark this answers is not in the document]
Comment (LumaC0) [reply src/lib/handlers/alks-developer-configure.ts:65]: Also worth noting that even though the order of configuration options has changed, the e2e test Except programs shouldn't have to be modified. Except runs through each pattern sequentially when new output arrives. I wrote about that along with other useful information in the scripts themselves.

[the remark this answers is not in the document]
Comment (elliottzack429) [reply src/lib/handlers/alks-developer-configure.ts:65]: Yep.  Ben thought of that one. Only thing is that having it there made it easier to fall back to basic auth.  But then we figured forcing them to update the refresh token was better.  They can still set it by doing alks developer login, at which point the fallback to basic auth would work again.

**Review by americk0 — COMMENTED** [review]
Code looks good to me. Once we get those checks passing I'll take another look and approve

**Review by americk0 — APPROVED** [review]
Looks like all the checks are passing except veracode (for reasons unrelated to your PR) so 👍

**Pull request merged**